pkg/ottl/ottlfuncs/func_sha1.go

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"crypto/sha1" // #nosec
+	"encoding/hex"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type SHA1Arguments[K any] struct {
+	Target ottl.StringGetter[K] `ottlarg:"0"`
+}
+
+func NewSHA1Factory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("SHA1", &SHA1Arguments[K]{}, createSHA1Function[K])
+}
+
+func createSHA1Function[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*SHA1Arguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("SHA1Factory args must be of type *SHA1Arguments[K]")
+	}
+
+	return SHA1HashString(args.Target)
+}
+
+func SHA1HashString[K any](target ottl.StringGetter[K]) (ottl.ExprFunc[K], error) {
+
+	return func(ctx context.Context, tCtx K) (interface{}, error) {
+		val, err := target.Get(ctx, tCtx)
+		if err != nil {
+			return nil, err
+		}
+
+		if val == "" {
+			return val, nil
+		}
+
```

Comment:
```suggestion
```
I think we should also remove this check. I ran the following snippet in the Go playground and it seems that hashing an empty string doesn't return `""`.

```golang
	hash := sha1.New() // #nosec
	_, err := hash.Write([]byte(""))
	if err != nil {
		fmt.Println(err)
	}
	hashValue := hex.EncodeToString(hash.Sum(nil))
	fmt.Println(hashValue) // da39a3ee5e6b4b0d3255bfef95601890afd80709
```